File: core/src/ru/myitschool/dcrawler/math/DiceAction.java
package ru.myitschool.dcrawler.math;

import java.util.Random;

/**
 * Created by Voyager on 04.08.2017.
 */
public class DiceAction extends MathAction {

    private int dice;
    private Random random;

    public DiceAction(int dice) {
        this.dice = dice;
        random = new Random();
    }


    @Override
    public int act() {
        int result = 0;
        result += random.nextInt(dice) + 1;
        return result;
    }

    @Override
    public String getDescription() {
        return "d" + dice;
    }

    @Override
    public int max() {
        return dice;
    }

    @Override
    public int min() {
        return 1;
    }
}

File: core/src/com/wolg_vlad/dcrawler/math/DiceAction.kt
package com.wolg_vlad.dcrawler.math

import java.util.*

/**
 * Created by Voyager on 04.08.2017.
 */
class DiceAction(private val dice: Int) : MathAction() {
    private val random: Random
    override fun act(): Int {
        var result = 0
        result += random.nextInt(dice) + 1
        return result
    }

    override val description: String?
        get() = "d$dice"

    override fun max(): Int {
        return dice
    }

    override fun min(): Int {
        return 1
    }

    init {
        random = Random()
    }
}